packages/langium/src/parser/parser-builder-base.ts

```diff
@@ -88,6 +88,8 @@ function buildElement(ctx: RuleContext, element: AbstractElement, ignoreGuard =
         method = buildUnorderedGroup(ctx, element);
     } else if (isGroup(element)) {
         method = buildGroup(ctx, element);
+    } else if(isEndOfFile(element)) {
+        method = () => ctx.parser.consume(0, EOF, element);
```

Comment:
Suggestion: Don't use index `0` here. Chevrotain will throw in error in case we encounter a rule such as `Test: 'x' EOF | EOF;` use the rule context `consume` property instead.

Reply:
TBH that line was done by Copilot XD, I forgot to review it, because the tests were working. I am not sure, what to write there. I would expect something like `consume++; if(consume >= input.length) { ...return EOF token }`

Reply:
DONE